### PR TITLE
Adds cri-dockerd

### PR DESCRIPTION
**Pull request by Nino-K**
This change introduces the mirantis Cri-dockerd, as preparation for docker deprecation.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -32,7 +33,7 @@ mkimage:
 		.
 
 .PHONY: iso
-iso: nerdctl-$(NERDCTL_VERSION)-$(ARCH) qemu-$(QEMU_VERSION)-copying
+iso: nerdctd-$(NERDCTL_VERSION)-$(ARCH) qemu-$(QEMU_VERSION)-copying cri-dockerd-$(CRI_DOCKERD_VERSION)-$(ARCH)
```
Comment (jandubois) [inline Makefile:36]: Editor-malfunction?

```suggestion
iso: nerdctl-$(NERDCTL_VERSION)-$(ARCH) qemu-$(QEMU_VERSION)-copying cri-dockerd-$(CRI_DOCKERD_VERSION)-$(ARCH)
```

Comment (Nino-K) [reply Makefile:36]: Hmmm, I'm not sure what happened there 🤔

File: Makefile [hunk 2]
```diff
@@ -32,7 +33,7 @@ mkimage:
...
 	ALPINE_VERSION=$(ALPINE_VERSION) NERDCTL_VERSION=$(NERDCTL_VERSION) QEMU_VERSION=$(QEMU_VERSION) REPO_VERSION=$(REPO_VERSION) EDITION=$(EDITION) BUILD_ID=$(BUILD_ID) ARCH=$(ARCH) ARCH_ALIAS=$(ARCH_ALIAS) ./build.sh
```
Comment (jandubois) [inline Makefile:37]: You need to pass in the `CRI_DOCKERD_VERSION` to `build.sh`:

```suggestion
	ALPINE_VERSION=$(ALPINE_VERSION) CRI_DOCKERD_VERSION=$(CRI_DOCKERD_VERSION) NERDCTL_VERSION=$(NERDCTL_VERSION) QEMU_VERSION=$(QEMU_VERSION) REPO_VERSION=$(REPO_VERSION) EDITION=$(EDITION) BUILD_ID=$(BUILD_ID) ARCH=$(ARCH) ARCH_ALIAS=$(ARCH_ALIAS) ./build.sh
```

**Comment by jandubois** [comment]
Can you please squash commits and force-push?

**Review by jandubois — APPROVED** [review]
Thanks, LGTM and works as expected.

File: genapkovl-lima.sh [hunk 3]
```diff
@@ -240,6 +240,18 @@ if [ "${LIMA_INSTALL_SSHFS}" == "true" ]; then
     echo "sshfs" >> "$tmp"/etc/apk/world
 fi
 
+if [ "${LIMA_INSTALL_CRI_DOCKERD}" == "true" ]; then
+    mkdir -p "${tmp}/cri-dockerd"
+    tar xz -C "${tmp}/cri-dockerd" -f /home/build/cri-dockerd.tar.gz
+    chmod u+s "${tmp}/cri-dockerd/cri-dockerd"
```
Comment (jandubois) [inline genapkovl-lima.sh:246]: I realized that we don't need this, so please remove.

I understand that you just copied from the nerdctl section above. We mark those setuid so they can be invoked without `sudo`, but then RD doesn't seem to take advantage of that (yet?).

But `cri-dockerd` should always be started by `/etc/init.d/cri-dockerd` so is already root.

**Review by jandubois — CHANGES_REQUESTED** [review]
Please amend commit to remove the setuid line.

File: genapkovl-lima.sh (hunk 3 above)
Comment (Nino-K) [reply genapkovl-lima.sh:246]: That makes sense. Thanks

**Review by jandubois — APPROVED** [review]
LGTM

**Pull request merged**